src/gui/properties/speedplotview.h

```diff
@@ -86,19 +86,17 @@ class SpeedPlotView : public QGraphicsView
     void paintEvent(QPaintEvent *event) override;
 
 private:
-    enum PeriodInSeconds
+    class Averager
     {
-        MIN1_SEC = 60,
-        MIN5_SEC = 5 * 60,
-        MIN30_SEC = 30 * 60,
-        HOUR6_SEC = 6 * 60 * 60
-    };
-
-    enum PointsToSave
-    {
-        MIN5_BUF_SIZE = 5 * 60,
-        MIN30_BUF_SIZE = 10 * 60,
-        HOUR6_BUF_SIZE = 20 * 60
+    public:
+        Averager(int divider, boost::circular_buffer<PointData> &sink);
+        void push(const PointData &p);
+        bool isReady();
+    private:
+        int m_counter;
```

Comment:
I would move it below m_sink.

Reply:
For me there is no difference, so I moved it there. But I'm curios what a difference it makes for you?

Reply:
Just to have some logical order: immutable fields (so-called "instance parameters") goes first and then other fields.